object_model/artifact.py

```diff
@@ -365,57 +365,179 @@ def pull(self, manifest={}):
-        if os.path.exists(self.xp_state.versioningDirectory):
-            rmtree(self.xp_state.versioningDirectory)
-            os.mkdir(self.xp_state.versioningDirectory)
+    def peek(self, head=25, manifest={}, bindings = {}, func = lambda x: x):
+
+        self.xp_state.versioningDirectory = os.path.expanduser('~') + '/' + 'flor.d'
+
+        # Recreate the tmpexperiment directory
+        tmpexperiment = self.xp_state.tmpexperiment
+        if os.path.exists(tmpexperiment):
+            rmtree(tmpexperiment)
+            os.mkdir(tmpexperiment)
         else:
-            os.mkdir(self.xp_state.versioningDirectory)
+            os.mkdir(tmpexperiment)
 
-        # Update elements visited and literals array
         self.xp_state.visited = []
         util.activate(self)
+        userDefFiles = set(os.listdir()) - self.xp_state.ghostFiles
+        experimentName = self.xp_state.EXPERIMENT_NAME
 
-        if taip == "default":
-            try:
-                # Just pull once and put output in folder peek0 within versioning directory.
-                original_dir = os.getcwd()
-                dst = self.xp_state.versioningDirectory + '/' + "peek0"
-                copytree(os.getcwd(), dst, True)
-                os.chdir(dst)
-                self.__pull__()
+        try:
+            old_literal_bindings = {}
+
+            # Iterate through bindings, store old bindings and reset to new bindings for pull
+            for literal, binding in bindings.items():
+                if literal.__oneByOne__:
+                    old_literal_bindings[literal] = literal.v[0]
+                    literal.v[0] = binding
+                else:
+                    old_literal_bindings[literal] = literal.v
+                    literal.v = binding
+
+
+            # If there are literals that are unbound, then set them to default.
+            if len(bindings) != len(self.__getLiteralsAttached__()):
+                unbound_literals = [literal for literal in self.xp_state.literals if literal not in bindings.keys()]
+                for literal in unbound_literals:
+                    if literal.__oneByOne__:
+                        old_literal_bindings[literal] = literal.v[0]
+                        literal.v[0] = literal.getDefault()
+                    else:
+                        old_literal_bindings[literal] = literal.v
+                        literal.v = literal.getDefault()
+
+            util.activate(self)
 
-                # dst = self.xp_state.versioningDirectory + '/' + "peek0"
-                # copytree(os.getcwd(), dst, True)
+            # Pull appropriate literal and create trial directory
+            self.__pull__()
 
-                # Unpickle and read the file to output.
-                if util.isPickle(dst + "/" + self.loc):
-                    out = func(util.unpickle(dst + '/' + self.loc))
+            # Write the config file
+            config = {}
+            for litName in self.literalNamesAddendum:
```

Comment:
where was literalNamesAddendum set?

Reply:
We are using this for pull in order to write the config file. Do we want to be writing a config file within peek?